packages/frontend-core/src/fetch/DataFetch.js

```diff
@@ -365,7 +364,9 @@ export default class DataFetch {
     let refresh = false
     const entries = Object.entries(newOptions || {})
     for (let [key, value] of entries) {
-      if (JSON.stringify(value) !== JSON.stringify(this.options[key])) {
+      const oldVal = this.options[key] == null ? null : this.options[key]
+      const newVal = value == null ? null : value
```

Comment:
Added this in as `undefined` and `null` were treating as being different and therefore causing a fetch to refresh all data.

Reply:
Nice catch!